NYPL-discovery/discovery-front-end
# Scc 3671

**What's this do?**
Update to ensure the new LPA locations (changing `my*` to `lp*` and `pa*`) don't break the lookup for research center hours and address.

Also simplifies the mechanism:
 - Remove the static `locationCodes` lookup, replacing it with a more flexible location id prefix check
 - Remove questionpoint URL lookups because it was tied into the deprecated lookup and we're no longer using those URLs
 - Updated README to reflect when/why these lookups need to be updated as locations change

**Why are we doing this? (w/ JIRA link if applicable)**
https://jira.nypl.org/browse/SCC-3671

**Do these changes have automated tests?**
Some

**How should this be QAed?**
We should confirm items in a few different research centers and ReCAP look right on the HoldRequest page.

This is now deployed on Train, so you can see [on a page like this](https://discoveryui-edd-training.nypl.org/research/research-catalog/hold/request/b13578104-i16551564?searchKeywords=breakfast) that an LPA location appears.

**Dependencies for merging? Releasing to production?**
This is built alongside updates to NYPL-Core (v2.03), but includes support for older NYPL-Core versions (since it supports the deprecated `my*` LPA location prefix). QA will confirm, but this should be safe to deploy to production regardless of NYPL-Core change rollout.

**Has the application documentation been updated for these changes?**
Yes

**Did someone actually run this code to verify it works?**
Me
